modules/google-analytics/classes/class-jetpack-google-amp-analytics.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Add support for Google Analytics e-commerce events for AMP pages.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Jetpack_Google_AMP_Analytics class.
+ */
+class Jetpack_Google_AMP_Analytics {
+	/**
+	 * Add hooks.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_add_to_cart', array( $this, 'amp_add_to_cart' ), 10, 6 );
+		add_action( 'woocommerce_thankyou', array( $this, 'amp_after_purchase' ), 10, 1 );
+		add_action( 'wp_footer', array( $this, 'amp_send_ga_events' ) );
+	}
+
+	/**
+	 * Generate a GA event when adding an item to the cart.
+	 *
+	 * @param string $cart_item_key Cart item key.
+	 * @param string $product_id Product ID.
+	 * @param int    $quantity Product quantity.
+	 * @param int    $variation_id Product variation ID.
+	 * @param object $variation Product variation.
+	 * @param object $cart_item_data Cart item data.
+	 */
+	public function amp_add_to_cart( $cart_item_key, $product_id, $quantity, $variation_id, $variation, $cart_item_data ) {
+		if ( ! class_exists( 'Jetpack_AMP_Support' ) || ! Jetpack_AMP_Support::is_amp_request() ) {
+			return;
+		}
+
+		$product = wc_get_product( $product_id );
+		if ( $product ) {
+			$product_sku  = Jetpack_Google_Analytics_Utils::get_product_sku_or_id( $product );
+			$product_name = $product->get_name();
+
+			$events   = WC()->session->get( 'wc_ga_events' );
+			$events[] = array(
+				'type'      => 'add',
+				'ga_params' => array(
+					'pa'    => 'add',
+					'pr1id' => $product_sku,
+					'pr1nm' => $product_name,
+					'pr1qt' => $quantity,
+				),
+			);
+			WC()->session->set( 'wc_ga_events', $events );
+		}
+	}
+
+	/**
+	 * Generate a GA event when removing an item to the cart.
+	 *
+	 * @param int $order_id The Order ID.
+	 */
+	public function amp_after_purchase( $order_id ) {
+		if ( ! class_exists( 'Jetpack_AMP_Support' ) || ! Jetpack_AMP_Support::is_amp_request() ) {
+			return;
+		}
+
```

Comment:
Like above, should we check for `Jetpack_Google_Analytics_Options::track_remove_from_cart_is_enabled()` here?